src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -191,7 +209,7 @@ private ParsedCondition(String keyword, boolean supportsRegex, String line) thro
         }
 
         @Override
-        public abstract boolean matches(IRoutingTile tile, EntityMinecart cart);
+        public abstract int evaluate(IRoutingTile tile, EntityMinecart cart);
```

Comment:
Instead of changing matches() and spamming valueForBool() everywhere, why not just have evaluate() wrap matches()?

This also falls into the fundamental flaws I think. While its nice that Conditions can return different values, I'm not sure how desirable that is from a complexity standpoint. Like having a Condition called "InventoryFullness" that returns the comparator value for an inventory would seem to me to be shoehorning an unrelated feature into an otherwise clean Boolean logic language.

Mapping TRUE to a arbitrary return value is one thing, actually making conditions that return numbers natively I think is not a great idea.

So I'd say find some other way to make this happen rather than modifying Condition.